client/purplecaffeine/core.py

```diff
@@ -143,15 +146,20 @@ def add_tag(self, tag: str):
 
     def save(self):
         """Save into Backend."""
-        self.backend.save(name=self.name, trial=self)
+        self.backend.save(trial=self)
 
-    def read_trial(self) -> "Trial":
+    def read_trial(self, trial_id: Optional[int] = None) -> Trial:
         """Read a trial from Backend.
 
+        Args:
+            trial_id: if backend is the remote api, you need the trial id find in database.
+
         Returns:
             Trial dict object
         """
-        return self.backend.get(name=self.name)
+        if isinstance(self.backend, LocalBackend):
+            return self.backend.get(name=self.name)
+        return self.backend.get(trial_id=trial_id)
```

Comment:
@IceKhan13 with our design I don't see another way to do 🤔  or do you have idea ?

Reply:
hm. maybe we should just query by id and not use name.

```python
return self.backend.get(trail_id)
```

Reply:
Ok but id works only for remote api and not for local

Reply:
why? 

We can save each trial as separate file and give it name `<storage_path>/purple_caffeline/<trial_id>.json`

and now for `LocalBackend` you will be able to do
```python
class LocalBackend(BaseBackend):
    ...
    def get(self, trial_id: str):
        file_path = f"{self.path}/purple_caffeline/{trial_id}.json"
        if os.path.exists(file_path):
            with open(file_path, "r") as file:
                return Trial(**json.load(file))
```

Reply:
I was meant for usage ^^'

In local, don't we want to have something browsable by name ? So it can be easier to know which file does what (without open them).
If one day, the user want to create a new experiment with the same name that gonna shut an error. But the user can manage its own files and wouldn't have problem renaming them.

Reply:
Or, I have an idea.

Another use case we could have is people sharing experiments between each other. So what i can do : it's create a an `import_external_experiments()` function which allow the user to create a Trial from any experiments file using the name of the file.
And so that make available a browsing using a name and not an id.